src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,98 @@
+package duke.parser;
+
+import duke.data.TaskHandler;
+import duke.data.task.Task;
+import duke.data.exception.DukeException;
+
+import duke.ui.Ui;
+import duke.command.*;
+import duke.storage.Storage;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Class that makes sense of the user commands to Duke.
+ *
+ * @author Won Ye Ji
+ */
+public class Parser {
+    static boolean isTerminated = false;
+    private TaskHandler taskHandler;
+    private Storage storage;
+
+    /**
+     * Initialise variables to run Duke.
+     */
+    public void initialiseDuke() {
+        try {
+            storage = new Storage("./data/taskList.txt");
+            ArrayList<Task> list = storage.loadTasks();
+            taskHandler = new TaskHandler(list);
+        } catch (DukeException e) {
+            System.err.println("Error: Unable to initialise duke");
+            System.out.println(e.getMessage());
+        }
+    }
+
+    /**
+     * Runs Duke.
+     */
+    public void runDuke() {
+        Ui.greet();
+        Scanner sc = new Scanner(System.in);
+        while(!isTerminated) {
+            String cmd = sc.nextLine();
+            String commandWord = cmd.split(" ")[0].toUpperCase();
+            try {
+                switch (commandWord) {
+                case ListCommand.COMMAND_WORD:
+                    ListCommand lc = new ListCommand(taskHandler, storage);
+                    lc.execute(cmd);
+                    break;
+
```

Comment:
I dont think a newline is needed after a `break` statement? Same for all the other cases below.
```suggestion

```